### PR TITLE
twoFactor enable/disable

### PR DESCRIPTION
**Pull request by mattlockyer**
Pre-requisite: https://github.com/near/near-api-js/pull/445

### CODE REVIEW

**Comment by vgrichina** [comment]
> To set up and use dependency branch of near-api-js: near-api-js/ms-disable

I think best to reder to specific branch  in `package.json` in such case (so that all deploys can still be tested easily), but then remove the change after near-api-js release

**Comment by mattlockyer** [comment]
> > To set up and use dependency branch of near-api-js: near-api-js/ms-disable
> 
> I think best to reder to specific branch in `package.json` in such case (so that all deploys can still be tested easily), but then remove the change after near-api-js release

https://github.com/near/near-wallet/blob/2523dd3d157f03293c8f393e93d579b6c32ebbc3/package.json#L13

**Review by kcole16 — APPROVED** [review]

File: src/utils/wallet.js [hunk 1]
```diff
@@ -819,7 +818,7 @@ class Wallet {
             const keyPair = KeyPair.fromString(secretKey)
             await tempKeyStore.setKey(NETWORK_ID, accountId, keyPair)
             account.keyStore = tempKeyStore
-            account.inMemorySigner = new nearApiJs.InMemorySigner(tempKeyStore)
+            account.inMemorySigner = account.connection.signer = new nearApiJs.InMemorySigner(tempKeyStore)
```
Comment (vgrichina) [inline src/utils/wallet.js:821]: `account.connection.signer = `

why is this needed? This doesn't look good to me. Why not create temporary `Account` instance  with necessary config instead of patching existing one?

Comment (mattlockyer) [reply src/utils/wallet.js:821]: was always confused by this:
https://github.com/near/near-wallet/blob/2fea47b429f844af95a4b21c60a0e081369575e1/src/utils/wallet.js#L629

Comment (vgrichina) [reply src/utils/wallet.js:821]: looks like it should have been `this.inMemorySigner` (field in `Wallet` object)

**Comment by vgrichina** [comment]
@mattlockyer when I try to disable 2FA this crashes here:

```
    async disableMultisig() {
        const contractBytes = new Uint8Array(await (await fetch('/main.wasm')).arrayBuffer())
        const result = await this.disable(contractBytes)   // <<<<<<<<<<<<<<<<<<<<
        await store.dispatch(refreshAccount())
        return result
    }
```

(there is no `.disable` method)

**Comment by mattlockyer** [comment]
> (there is no `.disable` method)

Can you check which version of near-api-js you're running? Are you linked to one locally?

I just disabled/enabled using https://near-wallet-pr-1107.onrender.com/ worked fine. Cache?

**Review by vgrichina — CHANGES_REQUESTED** [review]
has dangerous bugs blocking release

**Comment by vgrichina** [comment]
```
[I] ➜ NODE_ENV=testnet near keys berries.testnet
Keys for account berries.testnet
[
  {
    public_key: 'ed25519:5z2QWPyCmw69ic5vgPHLGoGCpHzMJ8PLS2NadEvKnWzV',
    access_key: { nonce: 0, permission: 'FullAccess' }
  },
  {
    public_key: 'ed25519:7Ywn6QxxYLhbSRTtiJupssWZ42bbxQvQ7Pa9YVWpnFzm',
    access_key: { nonce: 0, permission: 'FullAccess' }
  },
  {
    public_key: 'ed25519:GLwGYLqFq2wEnakLj3gwoH2jGwKqKARBuCYtqbZcfVLs',
    access_key: { nonce: 1, permission: 'FullAccess' }
  },
  [length]: 3
]
```

looks like after disabling 2FA my limited access key got replaced by FullAccess key

**Comment by vgrichina** [comment]
Screen Shot 2020-11-11 at 3 26 23 PM

re-enabling 2FA after disable gave error but seemingly succeeded

**Comment by vgrichina** [comment]
Needs more careful testing (including regression tests in near-api-js) to proceed. Need to make sure works fine with app logins as well (i.e. doesn't touch app keys).

**Comment by vgrichina** [comment]
@mattlockyer not a blocker for this fix, but "2fa code expired" error from server doesn't seem to be handled in UI yet, so kinda hard to understand what is going on in that case.

**Comment by vgrichina** [comment]
Also "Resend Code" should be clearing code, as you'll need to enter new one
